File: 백준/Silver/10845. 큐/큐.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? '/dev/stdin' : 'input.txt';
const input = fs.readFileSync(filePath).toString().trim().split('\n');
const orderArr = input.slice(1).map((i) => i.split(' '));
const arr = [];
let answer = [];

for (let i of orderArr) {
  let order = i[0];

  switch (order) {
    case 'push':
      arr.push(parseInt(i[1]));
      break;
    case 'pop':
      arr.length > 0 ? answer.push(arr.shift()) : answer.push(-1);
      break;
    case 'size':
      answer.push(arr.length);
      break;
    case 'empty':
      arr.length > 0 ? answer.push(0) : answer.push(1);
      break;
    case 'front':
      arr.length > 0 ? answer.push(arr[0]) : answer.push(-1);
      break;
    case 'back':
      arr.length > 0 ? answer.push(arr[arr.length - 1]) : answer.push(-1);
  }
}

console.log(answer.join('\n'));

File: 백준/Silver/10845. 큐/큐.py
N = int(input())
queue = []

for _ in range(N):
    inp = input().split()
    o = inp[0]

    if o == 'push':
        queue.append(int(inp[1]))
    elif o == 'pop':
        if queue:
            print(queue.pop(0))
        else:
            print(-1)
    elif o == 'size':
        print(len(queue))
    elif o == 'empty':
        if queue:
            print(0)
        else:
            print(1)
    elif o == 'front':
        if queue:
            print(queue[0])
        else:
            print(-1)
    elif o == 'back':
        if queue:
            print(queue[-1])
        else:
            print(-1)